### PR TITLE
Add graph debug doc

### PR DESCRIPTION
**Pull request by lixiang007666**
This PR is done:

- [x] Add debug() cn doc for Graph.
     Related PR：https://github.com/Oneflow-Inc/oneflow-documentation/pull/400
     Issue：https://github.com/Oneflow-Inc/oneflow-documentation/issues/386
image
image
image
image
image

### CODE REVIEW

**Comment by doombeaker** [comment]
CI 通过需要有 'cn' 或者 'en' 标签。主要是方便之后安排相关人根据 PR 去同步文档（比如先写的中文文章，然后后续要同步英文文档，除了看网站内容外，相关同事还可以通过筛选 cn 标签，看哪些是中文文档更新过的）

**Comment by doombeaker** [comment]
#400 关了是不再维护了吗。这个可以在  PR 400 或者这个 PR 里说明下么。因为后续会安排同学根据这些中文 PR 翻译更新英文版本的文档，怕混乱。

File: cn/docs/basics/08_nn_graph.md [hunk 1]
```diff
@@ -253,7 +253,7 @@ class GraphMobileNetV2(flow.nn.Graph):
 
 ### Graph 调试
 
-可以调用 `print` 打印 Graph 对象，输出 Graph 对象的信息。
+当前输出Graph的调试信息共有两种方式。**第一种** 可以调用 `print` 打印 Graph 对象，输出 Graph 对象的信息。
```
Comment (doombeaker) [inline cn/docs/basics/08_nn_graph.md:256]:
```suggestion
当前输出 Graph 的调试信息共有两种方式。**第一种** 可以调用 `print` 打印 Graph 对象，输出 Graph 对象的信息。
```

File: cn/docs/basics/08_nn_graph.md [hunk 2]
```diff
@@ -288,6 +288,8 @@ print(graph_mobile_net_v2)
 )
 ```
 
+在上面的调试信息中，表示基于 Sequential 模型，网络中自定义了 `ConvBNActivation` (对应 `MBConv` 模块)、卷积层(包括详细的`channel`、`kernel_size`和`stride`等参数信息)、`Dropout` 和全连接层等结构。
```
Comment (doombeaker) [inline cn/docs/basics/08_nn_graph.md:291]:
```suggestion
在上面的调试信息中，表示基于 Sequential 模型，网络中自定义了 `ConvBNActivation` (对应 `MBConv` 模块)、卷积层(包括详细的 `channel`、`kernel_size` 和 `stride` 等参数信息)、`Dropout`  和全连接层等结构。
```

File: cn/docs/basics/08_nn_graph.md [hunk 3]
```diff
@@ -315,7 +317,17 @@ print(graph_mobile_net_v2)
     ...
 ```
 
-此外，调用 Graph 对象的 [debug](https://oneflow.readthedocs.io/en/master/graph.html#oneflow.nn.Graph.debug) 方法，就开启了 Graph 的调试模式。
+**第二种** 方式是调用 Graph 对象的 [debug](https://oneflow.readthedocs.io/en/master/graph.html#oneflow.nn.Graph.debug) 方法，就开启了 Graph 的调试模式。
```
Comment (doombeaker) [inline cn/docs/basics/08_nn_graph.md:320]:
```suggestion
**第二种** 方式是调用 Graph 对象的 [debug](https://start.oneflow.org/oneflow-api-cn/graph.html#oneflow.nn.Graph.debug) 方法，就开启了 Graph 的调试模式。
```

中文版的文档是不是引用中文版的 API 文档（现在还没合并，不过很快就会合并）就可以了，然后之后的英文版文档，就引用英文版的 API 文档

File: cn/docs/basics/08_nn_graph.md [hunk 4]
```diff
@@ -315,7 +317,17 @@ print(graph_mobile_net_v2)
...
+
+```python
+graph_mobile_net_v2.debug(v_level=1) #v_level参数默认值为0
```
Comment (doombeaker) [inline cn/docs/basics/08_nn_graph.md:323]:
```suggestion
graph_mobile_net_v2.debug(v_level=1) # v_level 参数默认值为 0
```

**Comment by lixiang007666** [comment]
> #400 关了是不再维护了吗。这个可以在 PR 400 或者这个 PR 里说明下么。因为后续会安排同学根据这些中文 PR 翻译更新英文版本的文档，怕混乱。

好的。

File: cn/docs/basics/08_nn_graph.md [hunk 5]
```diff
@@ -327,14 +339,28 @@ OneFlow 在编译生成计算图的过程中会打印调试信息，比如，将
...
-还可以通过设置 `v_level` 参数，调整 `debug` 的输出详细程度：
+当前可以使用 `v_level` 选择详细调试信息级别，默认级别为 0，最大级别为 3。
 
-```python
-graph_mobile_net_v2.debug(v_level=1)  # 输出详细信息
-```
+- v_level=0 时，只输出最基础的警告和构图阶段信息，如构图时间。
+- v_level=1 时，将额外打印每个 `nn.Module` 的构图信息，具体内容在下面的表格中介绍。
+- v_level=2 时，在构图阶段，将额外打印每个 `Op` 的创建信息，包括名称、输入内容、设备和 `SBP` 信息等。
+- v_level=3 时，将额外打印每个 `Op` 更详细的信息，如 `loc` 信息方便问题代码定位。
```
Comment (doombeaker) [inline cn/docs/basics/08_nn_graph.md:347]:
```suggestion
- `v_level=0` 时，只输出最基础的警告和构图阶段信息，如构图时间。
- `v_level=1` 时，将额外打印每个 `nn.Module` 的构图信息，具体内容在下面的表格中介绍。
- `v_level=2` 时，在构图阶段，将额外打印每个 Op 的创建信息，包括名称、输入内容、设备和 SBP 信息等。
- `v_level=3` 时，将额外打印每个 Op 更详细的信息，如与代码位置有关的信息，方便定位代码问题。
```

某些内容是代码，建议加上反引号
某些不是代码内容，只是概念的，建议去掉反引号。`loc` 这个过于细致和具体，我就建议去掉了。

不过以上只是建议。

File: cn/docs/basics/08_nn_graph.md [hunk 6]
```diff
@@ -327,14 +339,28 @@ OneFlow 在编译生成计算图的过程中会打印调试信息，比如，将
...
+
+此外，为了开发者对 Graph 对象下的类型有更清晰的认知，下面对 `debug` 输出的内容进行分析，基本包括 `GRAPH`、`CONFIG`、`MODULE`、`PARAMETER`、`BUFFER`、`INPUT` 和 `OUTPUT` 七个等级的标签。
```
Comment (doombeaker) [inline cn/docs/basics/08_nn_graph.md:349]:
```suggestion
此外，为了开发者对 Graph 对象下的类型有更清晰的认知，下面对 `debug` 输出的内容进行分析，基本包括 `GRAPH`、`CONFIG`、`MODULE`、`PARAMETER`、`BUFFER`、`INPUT` 和 `OUTPUT` 七个类别的标签。
```

应该是 “类别”，不是“等级”？ 用等级的话容易和上面讲的 `v_level` 混淆了

File: cn/docs/basics/08_nn_graph.md [hunk 7]
```diff
@@ -327,14 +339,28 @@ OneFlow 在编译生成计算图的过程中会打印调试信息，比如，将
...
+
+|      Name      |                             Info                             |                           Example                            |
+| :------------: | :----------------------------------------------------------: | :----------------------------------------------------------: |
+|     GRAPH      |    用户所定义的 Graph 信息，依次是类型：名字：构造方法。     |        `(GRAPH:GraphMobileNetV2_0:GraphMobileNetV2)`         |
+|     CONFIG     | Graph 的配置信息。如是否处于训练模式，`training=True` 表示 Graph 处于训练模式，如果在 Graph 的预测模式，则对应 `training=False`。 |        `(CONFIG:config:GraphConfig(training=True, )`         |
+|     MODULE     | 对应 `nn.Module` ，MODULE 可以在 Graph 标签下层，同时，多个 MODULE 之间也存在层级关系。 | `(MODULE:model:MobileNetV2())`，其中，`MobileNetV2` 为用户复用 Eager 模式下的 Module 类名。 |
+|   PARAMETER    | 对比 MODULE 给出了更清晰的 weight 和 bias 信息，tensor 的信息可以在这里被发现。在构图时，tensor 的数据内容不太重要，所以只展示了核心信息，这些信息对构建网络更为重要。此外，池化层等往往没有该内容。 | `(PARAMETER:model.features.0.1.weight:tensor(..., device='cuda:0', size=(32,), dtype=oneflow.float32, requires_grad=True))` |
```
Comment (doombeaker) [inline cn/docs/basics/08_nn_graph.md:356]:
```suggestion
|   PARAMETER    | 与 MODULE 给出了更清晰的 weight 和 bias 信息相比，这里更多地展现了 tensor 的信息。在构图时，tensor 的数据内容不太重要，所以只展示了核心信息（如形状、设备等），这些信息对构建网络更为重要。此外，池化层等往往没有该内容。 | `(PARAMETER:model.features.0.1.weight:tensor(..., device='cuda:0', size=(32,), dtype=oneflow.float32, requires_grad=True))` |
```

我的理解正确吗。其实我不是很理解 “对比 MODULE 给出了更清晰的 weight 和 bias 信息，tensor 的信息可以在这里被发现” 是不是和打印 Module 比的意思。如果是和 Module 的打印信息比较，那么我建议考虑下我的建议修改方式

File: cn/docs/basics/08_nn_graph.md [hunk 8]
```diff
@@ -327,14 +339,28 @@ OneFlow 在编译生成计算图的过程中会打印调试信息，比如，将
...
+|     BUFFER     |                在训练时产生的统计特性等内容。                | `(BUFFER:model.features.0.1.running_mean:tensor(..., device='cuda:0', size=(32,), dtype=oneflow.float32))` |
```
Comment (doombeaker) [inline cn/docs/basics/08_nn_graph.md:357]: “在训练时产生的统计特性等内容。”  我不了解，方便先 comment 里解释下不。之后看看有没有必要放入到这个 PR 里。

**Review by doombeaker — COMMENTED** [review]
主要 review 了中英文之间缺少空格的地方。以及个别我自己不太了解又想搞明白的地方。

File: cn/docs/basics/08_nn_graph.md (hunk 8 above)
Comment (lixiang007666) [reply cn/docs/basics/08_nn_graph.md:357]: 这个统计特性指：比如BN层里产生的running_mean和running_var信息。

File: cn/docs/basics/08_nn_graph.md (hunk 7 above)
Comment (lixiang007666) [reply cn/docs/basics/08_nn_graph.md:356]: 这里表述的有点问题，我修改下。

**Review by strint — APPROVED** [review]
